### PR TITLE
Testing Refactor - Run tests in parallel (WIP)

### PR DESCRIPTION
**Pull request by jvanderhoof**
#### What does this pull request do?
* Moves test script mostly into `ci/test`
* Allows all tests except for API to be run outside the Conjur container
* Allows all tests to be run in parallel

#### What background context can you provide?
This is part of an effort to have a clean, standard approach to running all tests.

#### Where should the reviewer start?
`ci/test`

#### How should this be manually tested?
Tests can be run with:
* `./test --cucumber-authenticators` - runs authenticator cuke tests
* `./test --cucumber-policy` - runs policy cuke tests
* `./test --rspec` - runs RSpec tests

#### Link to build in Jenkins (if appropriate)
https://jenkins.conjur.net/job/cyberark--conjur/job/authn-ldap-cucumber-refactor/

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -44,17 +44,40 @@ pipeline {
...
+          steps { sh 'cd ci && ./test --rspec' }
+        }
+        stage('Authenticators') {
+          steps { sh 'cd ci && ./test --cucumber-authenticators' }
+        }
+        stage('Policy') {
+          steps { sh 'cd ci && ./test --cucumber-policy' }
+        }
+        stage('API') {
+          steps { sh './test.sh' }
```
Comment (apotterri) [inline Jenkinsfile]: This could maybe use a comment (with TODO) indicating why it's different from the others.

File: lib/tasks/dev/jenkins.rb [hunk 2]
```diff
@@ -49,7 +49,5 @@
 
 task jenkins: [
   'db:migrate',
-  'jenkins:core:rspec',
   'jenkins:core:cucumber-api',
-  'jenkins:core:cucumber-policy'
```
Comment (apotterri) [inline lib/tasks/dev/jenkins.rb]: I'd get rid of the definitions of these tasks, too.

File: ci/test [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+# Setup to allow compose to run in an isolated namespace
+export COMPOSE_PROJECT_NAME="$(openssl rand -hex 3)"
+
+services="pg conjur"
+RUN_ALL=true
+RUN_AUTHENTICATORS=false
+# RUN_API=false # Need to refactor API tests for this to work
+RUN_POLICY=false
+RUN_RSPEC=false
+while true ; do
```
[the remark this answers is not in the document]
Comment (jvanderhoof) [reply ci/test]: After a bunch of attempts, I'm going to stick with this approach.  `getopt` support is questionable at best (optimal in a linux bash shell): https://stackoverflow.com/questions/12152077/how-can-i-make-bash-deal-with-long-param-using-getopt-command-in-mac.  `getopts` is widely available, but only works for single character arguments.

File: cucumber/policy/features/support/world.rb [hunk 4]
```diff
@@ -73,7 +73,9 @@ def admin_api_key
   end
 
   def admin_password
-    ENV['CONJUR_AUTHN_PASSWORD'] || 'admin'
+    ENV['CONJUR_AUTHN_API_KEY'] || begin
+      raise StandardError.new('Environment variable `CONJUR_AUTHN_API_KEY` must be set.')
+    end
```
Comment (apotterri) [inline cucumber/policy/features/support/world.rb]: Maybe you took my comment too literally. :)

I was thinking something more along the lines of

`raise 'Provide CONJUR_AUTHN_API_KEY' unless ENV['CONJUR_AUTHN_API_KEY'].present?`

Either way is fine, though.

Comment (jvanderhoof) [reply cucumber/policy/features/support/world.rb]: I was originally thinking what you were thinking, then I started googling.... You know where that ends 😀

**Pull request merged**